tests/pipeline/test_pipeline_with_transcoding.py

```diff
@@ -177,6 +181,24 @@ def test_duplicates_in_transcoded_outputs(self):
                 ]
             )
 
+    def test_transcoding_loop(self):
+        with pytest.raises(CircularDependencyError, match="node1"):
+            modular_pipeline(
+                [
+                    node(identity, "A@pandas", "B@pandas", name="node1"),
+                    node(identity, "B@spark", "C@spark", name="node2"),
+                    node(identity, "C@spark", "A@spark", name="node3"),
+                ]
+            )
+
+    def test_transcoding_self_reference(self):
```

Comment:
This test won't be needed after #3812 merge, so just pointing here not to forget about that